lib/elixir/test/elixir/file_test.exs

```diff
@@ -1322,6 +1322,26 @@ defmodule FileTest do
     assert Enum.count(stream) == 2
   end
 
+  test "stream keeps BOM" do
+    src = fixture_path("utf8_bom.txt")
+    bom_line = src
+    |> File.stream!()
+    |> Enum.take(1)
+    |> List.first()
```

Comment:
+1 for `[bom_line]`. I'd rather wrap something with list, than argue how to extract value from it :).

@whatyouhide why `hd/1` is better than List functions?